.github/workflows/mobilecoin-dev-cd.yaml

```diff
@@ -12,9 +12,10 @@ env:
 on:
   push:
     branches:
-    - release/v*
     - feature/*
-    - develop
+    - release/*
```

Comment:
`develop` got removed here, we're currently using `develop` in the [sgx](https://github.com/mobilecoinfoundation/sgx) repo.  I'm guessing this may be CD versus CI usage.

Reply:
We're not using develop in this repo, so pulling this out.